core/src/main/java/org/lflang/federated/extensions/CExtensionUtils.java

```diff
@@ -100,7 +100,12 @@ public static String initializeTriggersForNetworkActions(
    */
   public static String stpStructs(FederateInstance federate) {
     CodeBuilder code = new CodeBuilder();
-    federate.staaOffsets.sort((d1, d2) -> (int) (d1.time - d2.time));
```

Comment:
Nice catch, this was problematic if d2.time is FOREVER?

Reply:
Yes, this sort function would not work with forever. I've updated the new test to use forever.